From 8847a6a4c5bdafd5b5486cf4b35a8792fd33e332 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sat, 22 Jun 2019 13:05:32 +0530
Subject: [PATCH 086/138] octeontx2-af: Create tx_stall debugfs entry only for
 96xx

commit b5ae8c39c9cf0e47073353f3c945911193231fc5 from
git@git.assembla.com:cavium/WindRiver.linux.git

Silicons otherthan 96xx A0 doesn't have HW issues which result
in transmit stall, hence don't create tx_stall debugfs entry.

Change-Id: I9903d344634c6fac06cafd6315b2ffac93b49d42
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11461
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/rvu_debugfs.c   | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
index c7ffe6c02720..af4706017bbd 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
@@ -1647,10 +1647,13 @@ static void rvu_dbg_nix_init(struct rvu *rvu)
 	if (!pfile)
 		goto create_failed;
 
-	pfile = debugfs_create_file("tx_stall_hwissue", 0600, rvu->rvu_dbg.nix,
-				    rvu, &rvu_dbg_nix_tx_stall_hwissue_fops);
-	if (!pfile)
-		goto create_failed;
+	if (is_rvu_96xx_A0(rvu)) {
+		pfile = debugfs_create_file("tx_stall_hwissue", 0600,
+					    rvu->rvu_dbg.nix, rvu,
+					    &rvu_dbg_nix_tx_stall_hwissue_fops);
+		if (!pfile)
+			goto create_failed;
+	}
 
 	return;
 create_failed:
-- 
2.17.1

